### PR TITLE
resolves #11 use Iterable instead of Iterator

### LINKED ISSUE #11: Make generator implement Iterable instead of Iterator
Implementing Iterable will allow generators to work with spock in the same way but will allow for keeping data like size of collections intact.

### PR DESCRIPTION
**Pull request by Bijnagte**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -23,16 +23,16 @@ apply plugin: 'com.jfrog.bintray'
...
-version = '0.4.0'
+version = '0.5.0'
 
 repositories {
 	mavenCentral()
 }
 
 dependencies {
-  	compile('org.codehaus.groovy:groovy:2.4.5')
+  	compile('org.codehaus.groovy:groovy:2.4.6')
```
Comment (mariogarcia) [inline build.gradle]: Why not 2.4.7 ?

**Comment by mariogarcia** [comment]
I changed to this branch and it fails complaining about `groovydoc`

```
dev@bf958d774356:~/ws$ ./gradlew clean build
:clean
:compileJava UP-TO-DATE
:compileGroovy
:processResources
:classes
:jar
:groovydoc
Trying to override old definition of task fileScanner
:groovydoc FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':groovydoc'.
> groovydoc doesn't support the "noTimestamp" attribute

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 8.013 secs
```

It seems to be something related to latests versions of gradle (2.13 and 2.14). It seems `groovydoc` requires to use `groovy-all`. I think it's not big issue since we're already getting groovy-all as a spock transitive dependency. 

Bottom line, I would change 

`compile('org.codehaus.groovy:groovy:2.4.6')`

to this

`compile('org.codehaus.groovy:groovy-all:2.4.6')`

**Comment by Bijnagte** [comment]
Thanks @mariogarcia, I bumped the groovy version to the latest also

**Comment by mariogarcia** [comment]
It looks good to me. Do you want to add anything else ?

**Comment by Bijnagte** [comment]
probably not to this one, I have a few other things in mind for the next release. You mentioned finding a bug, is there something else that should be addressed?

**Comment by mariogarcia** [comment]
Yes, I forgot to add it. I've just done it. It has to do with `Gen.these`. Check it out and see if it you want to add it to this version. It's a bug, so you could always add a minor bug fixing release like `5.0.1`

**Pull request merged**